### PR TITLE
Move the assembly property information to the msbuild info article

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

This moves all of the content that was in [assembly-info.md](https://docs.microsoft.com/en-us/dotnet/core/migration/assembly-info) to the MSBuild article. There is a note in the linked issue to reference this content as part of the newer migration guide coming soon.

@gewarren I see that this file was recently created, however, I'm getting ready to start work on redoing all the migration content (with help from the product group) and I'm working on cleaning this stuff up. I think it makes more sense to have this information in the MSBuild props article and then referenced by migration articles.

Contributes to #22858

### CODE REVIEW

**Comment by gewarren** [comment]
@adegeo The content was intentionally split out (from the old csproj.md article) based on [this request](https://github.com/dotnet/docs/issues/14683). So to move it back in contradicts the intent of that issue, which was to improve SEO/findability. Do you think linking from the migration guide will solve that problem?

**Comment by adegeo** [comment]
@gewarren Ahhh gotcha. OK that makes sense why it was still here. However, I only saw 32 visitors in the last month, that can be considered such low traffic that the article isn't worth keeping around. I don't know if we're hitting our SEO targets for this article as it stands. Searching for various forms of `assemblyinfo csproj file` and `assemblyinfo csproj error` and `duplicate assembly attribute csproj` continually point me to StackOverflow Q/A and other articles. I couldn't surface this one.

- https://stackoverflow.com/questions/42138418/equivalent-to-assemblyinfo-in-dotnet-core-csproj
- https://stackoverflow.com/questions/10311347/duplicate-assemblyversion-attribute

Something else I noticed though too is that I think the MSBuild info article may be hurting overall SEO also since all of the H2s have no descriptions, only links under them. Anyway, with the low traffic, working on adding descriptions to the H2s in the MSBuild article, cross linking to it in further migration guides, we should really have all the bases covered.

Thoughts?

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
 </PropertyGroup>
 ```
 
+## Assembly info generation properties
+
+- [GenerateAssemblyInfo](#generateassemblyinfo)
+- [GeneratedAssemblyInfoFile](#generatedassemblyinfofile)
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:65]: Can you alphabetize this index?

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+Controls `AssemblyInfo` attribute generation for the project. The default value is `true`. Use `false` to disable generation of the file:
+
+```xml
+<PropertyGroup>
+  <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+</PropertyGroup>
+```
+
+The name of the file generated is controlled by the [GeneratedAssemblyInfoFile](#generatedassemblyinfofile) setting.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:87]:
```suggestion
The [GeneratedAssemblyInfoFile](#generatedassemblyinfofile) setting controls the name of the generated file.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 3]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+When this value is `true`, project properties are transformed into `AssemblyInfo` attributes. The following table lists the project properties that generate the attributes, and the property that can disable it:
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:89]:
```suggestion
When the `GenerateAssemblyInfo` value is `true`, project properties are transformed into `AssemblyInfo` attributes. The following table lists the project properties that generate the attributes, and the properties that can disable that generation:
```

File: docs/core/project-sdk/msbuild-props.md [hunk 4]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+| Property               | Attribute                                                      | Property to disable                                                                               |
+|------------------------|----------------------------------------------------------------|---------------------------------------------------------------------------------------------------|
+| `Company`              | <xref:System.Reflection.AssemblyCompanyAttribute>              | [`GenerateAssemblyCompanyAttribute`](#generateassemblycompanyattribute)                           |
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:93]: Should we add a line with a link to this section from https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#package-properties?

File: docs/core/project-sdk/msbuild-props.md [hunk 5]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+- `AssemblyVersion` and `FileVersion` default to the value of `$(Version)` without the suffix. For example, if `$(Version)` is `1.2.3-beta.4`, then the value would be `1.2.3`.
+- `InformationalVersion` defaults to the value of `$(Version)`.
+- If the `$(SourceRevisionId)` property is present, it's appended to `InformationalVersion`. You can disable this behavior using `IncludeSourceRevisionInInformationalVersion`.
+- `Copyright` and `Description` properties are also used for NuGet metadata.
+- `Configuration`, which defaults to `Debug`, is shared with all MSBuild targets. You can set it via the `--configuration` option of `dotnet` commands, for example, [dotnet pack](../tools/dotnet-pack.md).
+
+#### Migrating from .NET Framework
+
+.NET Framework project templates created a code file with these assembly info attributes set, typically located at *.\Properties\AssemblyInfo.cs* or *.\Properties\AssemblyInfo.vb*. Newer SDK-style projects generate this file for you based on the project settings. **You can't have both.** When porting your code to .NET 5 (and .NET Core 3.1) or later, do one of the following:
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:114]:
```suggestion
.NET Framework project templates create a code file with these assembly info attributes set. The file is typically located at *.\Properties\AssemblyInfo.cs* or *.\Properties\AssemblyInfo.vb*. SDK-style projects generate this file for you based on the project settings. **You can't have both.** When porting your code to .NET 5 (and .NET Core 3.1) or later, do one of the following:
```

File: docs/core/project-sdk/msbuild-props.md [hunk 6]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+- Disable the generation of temporary code file that contains the assembly info attributes by setting `GenerateAssemblyInfo` to `false`. This enables you to keep your *AssemblyInfo* file.
+- Migrate the settings in the `AssemblyInfo` file to the project, and delete the `AssemblyInfo` file.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:117]:
```suggestion
- Migrate the settings in the `AssemblyInfo` file to the project file, and delete the `AssemblyInfo` file.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 7]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+- `InformationalVersion` defaults to the value of `$(Version)`.
+- If the `$(SourceRevisionId)` property is present, it's appended to `InformationalVersion`. You can disable this behavior using `IncludeSourceRevisionInInformationalVersion`.
+- `Copyright` and `Description` properties are also used for NuGet metadata.
+- `Configuration`, which defaults to `Debug`, is shared with all MSBuild targets. You can set it via the `--configuration` option of `dotnet` commands, for example, [dotnet pack](../tools/dotnet-pack.md).
+
+#### Migrating from .NET Framework
+
+.NET Framework project templates created a code file with these assembly info attributes set, typically located at *.\Properties\AssemblyInfo.cs* or *.\Properties\AssemblyInfo.vb*. Newer SDK-style projects generate this file for you based on the project settings. **You can't have both.** When porting your code to .NET 5 (and .NET Core 3.1) or later, do one of the following:
+
+- Disable the generation of temporary code file that contains the assembly info attributes by setting `GenerateAssemblyInfo` to `false`. This enables you to keep your *AssemblyInfo* file.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:116]:
```suggestion
- Disable the generation of the temporary code file that contains the assembly info attributes by setting `GenerateAssemblyInfo` to `false`. This enables you to keep your *AssemblyInfo* file.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 8]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+- Migrate the settings in the `AssemblyInfo` file to the project, and delete the `AssemblyInfo` file.
+
+### GeneratedAssemblyInfoFile
+
+The relative or absolute path of the generated assembly info file. Defaults to a file named *[project-name].AssemblyInfo.[cs|vb]* in the `$(IntermediateOutputPath)` (usually the *obj*) directory.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:121]:
```suggestion
The property defines the relative or absolute path of the generated assembly info file. Defaults to a file named *[project-name].AssemblyInfo.[cs|vb]* in the `$(IntermediateOutputPath)` (usually the *obj*) directory.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 9]
```diff
@@ -59,6 +59,173 @@ Use the `NetStandardImplicitPackageVersion` property when you want to specify a
...
+
+```xml
+<PropertyGroup>
+  <GeneratedAssemblyInfoFile>false</GeneratedAssemblyInfoFile>
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:125]: I don't think `false` is correct in this example.

**Comment by gewarren** [comment]
> @gewarren Ahhh gotcha. OK that makes sense why it was still here. However, I only saw 32 visitors in the last month, that can be considered such low traffic that the article isn't worth keeping around. I don't know if we're hitting our SEO targets for this article as it stands. Searching for various forms of `assemblyinfo csproj file` and `assemblyinfo csproj error` and `duplicate assembly attribute csproj` continually point me to StackOverflow Q/A and other articles. I couldn't surface this one.

The article has only been live since January 8th 2021.

**Comment by adegeo** [comment]
@gewarren 

> The article has only been live since January 8th 2021.

Ahh OK that makes sense then why results haven't surfaced. Do you think it would be better to move this to an include and have it in both? Full reference in this article and then the good-for-SEO article?

**Comment by gewarren** [comment]
> Do you think it would be better to move this to an include and have it in both? Full reference in this article and then the good-for-SEO article?

I don't think that's necessary. I wonder, though, if we should consider breaking up msbuild-props.md into separate articles for each section, for SEO.

**Review by gewarren — APPROVED** [review]

**Comment by adegeo** [comment]
@gewarren regarding SEO. H2s should be SEO scannable, but I think it helps SEO to have a paragraph description right below the H2. I think the lack of text below the H2 hurts these sections.

**Pull request merged**